libcoap/src/types.rs

```diff
@@ -730,6 +737,7 @@ impl CoapUri {
     ) -> Result<CoapUri, UriParsingError> {
         let mut uri = Self::create_unparsed_uri(uri_str, is_proxy);
 
+        ensure_coap_started();
```

Comment:
Should `ensure_coap_started()` be moved to the first line here?

Reply:
```suggestion
        ensure_coap_started();
        let mut uri = Self::create_unparsed_uri(uri_str, is_proxy);

```